cmd/clusterctl/client/cluster/mover.go

```diff
@@ -1164,3 +1173,35 @@ func (o *objectMover) checkTargetProviders(toInventory InventoryClient) error {
 
 	return kerrors.NewAggregate(errList)
 }
+
+func patchTopologyManagedFields(ctx context.Context, oldManagedFields []metav1.ManagedFieldsEntry, obj *unstructured.Unstructured, cTo client.Client) error {
+	var containsTopologyManagedFields bool
+	// managedFields will contain the target managed fields obj gets patched to.
+	wantManagedFields := []metav1.ManagedFieldsEntry{}
+
+	for _, m := range oldManagedFields {
+		if m.Manager == "manager" && m.Operation == metav1.ManagedFieldsOperationUpdate && m.Subresource == "" {
```

Comment:
Added a suggestion below for more comments + godoc on the function.

For the filtering: 

> Filter out the entry which only contains ownership of an ownerReference which does not exist anymore because the uid of the owner changed during move.

This is optional and after rethinking: I think we should remove that because we also don't clean this up on non topology-managed fields to stay consistent. (It has no impact to the topology controller).

Reply:
Example for the managed field which gets filtered:
```yaml
apiVersion: controlplane.cluster.x-k8s.io/v1beta1
kind: KubeadmControlPlane
metadata:
  ...
  managedFields:
  # this one:
  - apiVersion: controlplane.cluster.x-k8s.io/v1beta1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:ownerReferences:
          k:{"uid":"820924df-401f-4621-aea2-c8c0f7af9e64"}: {}
    manager: manager
    operation: Update
    time: "2022-06-09T09:10:30Z"
  ...
  name: self-hosted-1jopja-jpp9c
  namespace: self-hosted-6inrog
  ownerReferences:
  - apiVersion: cluster.x-k8s.io/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Cluster
    name: self-hosted-1jopja
    uid: 25a732ef-2ed6-4097-9cc5-73de4682b864 # THE UID SHOULD MATCH HERE BUT DOES NOT ANYMORE
  resourceVersion: "1463"
  uid: 1215fcae-3307-407c-83d1-250922e213d7
spec:
  ...
```

Reply:
We dropped the filtering :-)

Reply:
That makes it a lot easier, thx! :)